shell/src/main/java/alluxio/cli/bundler/command/CollectAlluxioInfoCommand.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.bundler.command;
+
+import alluxio.client.file.FileSystemContext;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.AlluxioException;
+import alluxio.shell.CommandReturn;
+import alluxio.shell.ShellCommand;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Command to run a set of Alluxio commands.
+ * Collects information about the Alluxio cluster.
+ * */
+public class CollectAlluxioInfoCommand extends AbstractInfoCollectorCommand {
+  public static final String COMMAND_NAME = "runAlluxioCheck";
+  private static final Logger LOG = LoggerFactory.getLogger(CollectAlluxioInfoCommand.class);
+  private static final String OUTPUT_FILE_NAME = "alluxioInfo.txt";
+
+  private String mAlluxioPath;
+  private List<AlluxioCommand> mAlluxioCommands;
+
+  /**
+   * Creates a new instance of {@link CollectAlluxioInfoCommand}.
+   *
+   * @param fsContext the {@link FileSystemContext} to execute in
+   * */
+  public CollectAlluxioInfoCommand(@Nullable FileSystemContext fsContext) {
+    super(fsContext);
+    mAlluxioPath = Paths.get(fsContext.getClusterConf().get(PropertyKey.WORK_DIR), "bin/alluxio")
+            .toAbsolutePath().toString();
+    registerCommands();
+  }
+
+  /**
+   * A special shell command that runs an Alluxio cmdline operation.
+   * */
+  public static class AlluxioCommand extends ShellCommand {
+    String mName;
+    String mAlluxioPath;
+    ShellCommand mAlternative;
+
+    /**
+     * Creates an instance of {@link AlluxioCommand}.
+     *
+     * @param name name of this Alluxio operation
+     * @param alluxioPath where Alluxio can be found
+     * @param cmd Alluxio cmd to run
+     * @param alternative an alternative option if the Alluxio operation fails
+     *                    for reasons like the cluster being currently down
+     * */
+    public AlluxioCommand(String name, String alluxioPath, String cmd, ShellCommand alternative) {
+      super((alluxioPath + " " + cmd).split(" "));
```

Comment:
(nit) use `new String[] {alluxioPath, cmd}`?

Reply:
Actually `new String[] {alluxioPath, cmd}` doesn't work. Seems `bin/alluxio` wants the commands to be a string array rather than one big concatenated string.